word_movers_distance/Calculate_WMD_cos_scores.ipynb

```diff
@@ -0,0 +1,3817 @@
+      "  return w2v.wmdistance(a,b)\n"
+     ]
+    }
+   ],
+   "source": [
+    "pos_cult = get_sim_perspective(pos_cult)\n",
+    "neg_cult = get_sim_perspective(neg_cult)\n",
+    "\n",
+    "pos_demog = get_sim_perspective(pos_demog)\n",
+    "neg_demog = get_sim_perspective(neg_demog)\n",
+    "\n",
+    "pos_rela = get_sim_perspective(pos_rela)\n",
+    "neg_rela = get_sim_perspective(neg_rela)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "25usK4EBjXoM"
+   },
+   "source": [
+    "## Calculate cosine similarity score between the article and each perspective"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "LQTJDmbcj3-8"
+   },
+   "source": [
+    "Using Word2Vec to get vector representations of the text in each jstor article, average them to get the article embedding, then calculate cosine similarity between the embedding of each article and average w2v vector for each perspective."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "XJ2kEH8wjjF2"
+   },
+   "source": [
+    "reference: https://github.com/h2researchgroup/embeddings/blob/main/preprocessing/cosine_values_calculate_word2vec.py"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {
+    "id": "B1DZj9J3pf63"
+   },
+   "outputs": [],
+   "source": [
+    "cult_core_embs=[]\n",
+    "for x in cult_core:\n",
+    "  try:\n",
+    "    cult_core_embs.append(w2v.wv[x])\n",
+    "  except:\n",
+    "    pass\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {
+    "id": "iagTUAI2W4kK"
+   },
+   "outputs": [],
+   "source": [
+    "demog_core_embs=[]\n",
+    "for x in dem_core:\n",
+    "  try:\n",
+    "    demog_core_embs.append(w2v.wv[x])\n",
+    "  except:\n",
+    "    pass\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {
+    "id": "ciBztsAbYpJN"
+   },
+   "outputs": [],
+   "source": [
+    "rela_core_embs=[]\n",
+    "for x in rela_core:\n",
+    "  try:\n",
+    "    rela_core_embs.append(w2v.wv[x])\n",
+    "  except:\n",
+    "    pass\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {
+    "id": "gCPBxyzkY3dF"
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "rela_core_embs_mean = np.mean(rela_core_embs, axis=0)\n",
+    "cult_core_embs_mean = np.mean(cult_core_embs, axis=0)\n",
+    "demog_core_embs_mean = np.mean(demog_core_embs, axis=0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 41,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def doc_words_cosine(doc):\n",
+    "    \"\"\"\n",
+    "  doc = str of text\n",
+    "  get cosine similarity scores between the document embedding and average emb of the core terms in each perspective.\n",
+    "  \"\"\"\n",
+    "    ss = [doc.split()]\n",
+    "    bb = build_phrases(ss)\n",
```

Comment:
See above comment. This should use the phraser we already trained, not build a new one.

Reply:
@Nancy078 this code needs to be updated. I think best way is to use the existing phraser model under top level of `models_storage/`. But at least be consistent with how you prep texts for WMD--looks like this just doesn't phrase at all at this point.

Reply:
resolved in new pr